graph/GuidelinesGraph.md

```diff
@@ -0,0 +1,441 @@
+    services to supply a specific error code to help differentiate errors that
+    share the same top-level error code but reported for different reasons.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+{
+  "error": {
+    "code": "BadRequest",
+    "message": "Cannot process the request because it is malformed or incorrect.",
+    "innererror": {
+      "code": "requiredFieldOrParameterMissing",
+           
+    }
+  }
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+| Microsoft Graph enforces the following error rules                                                                    | 
+|-----------------------------------------------------------------------------------------------------------------------|
+| :heavy_check_mark: **MUST** return an error property with a child code property in all error responses.                 | 
+| :heavy_check_mark: **MUST** return a 403 Forbidden error when insufficient scopes are present in the auth token.        | 
+| :heavy_check_mark: **MUST** return a 429 Too many requests error when the caller has exceeded throttling limits.        | 
+| :ballot_box_with_check: **SHOULD** return a 404 Not found error if a 403 would result in information disclosure. |
+
+For a complete mapping of error codes to HTTP statuses you can refer to the
+[rfc7231 (ietf.org)](https://datatracker.ietf.org/doc/html/rfc7231#section-6).
+
+## API contract and non-backward compatible changes
+
+Microsoft Graph definition of breaking changes is based on the [Microsoft REST
+API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/graph/Guidelines.md#123-definition-of-a-breaking-change).
+In general, making all but additive changes to the API contract for existing elements is
+considered breaking. Adding new elements is allowed and not considered a
+breaking change.
+
+\*\* Non-breaking changes:\*\*
+
+
+-   Addition of properties that are nullable or have a default value
+-   Addition of a member to an evolvable enumeration 
+-   Removal, rename, or change to the type of an annotation   
+-   Changes to the order of properties
+-   Changes to the length or format of opaque strings, such as resource IDs
+-   Addition or removal of an annotation OpenType="true" 
+
+\*\* Breaking changes:\*\*
+
+-   Changes to the URL or fundamental request/response associated with a
+    resource
+-   Removal, rename, or change to an incompatible type of a declared property
+-   Removal or rename of APIs or API parameters
+-   Addition of a required request header
+-   Addition of a EnumType members for non-evolvable enumerations
+-   Addition of a Nullable="false" properties to existing types
+-   Addition of a Nullable="false" parameters to existing actions and functions
+-    Changes to top-level error codes
+-    Introduction of server-side pagination to existing collections
+-    Changes to the default order of collection elements
```

Comment:
do we guarantee this across workloads today?  will /users *always* return users in the same order, now and in the future?

Reply:
will remove based on the working group recommendation